mm2src/coins/utxo/qtum.rs

```diff
@@ -230,7 +243,7 @@ impl UtxoCommonOps for QtumCoin {
 
     fn denominate_satoshis(&self, satoshi: i64) -> f64 { utxo_common::denominate_satoshis(&self.utxo_arc, satoshi) }
 
-    fn my_public_key(&self) -> &Public { self.utxo_arc.key_pair.public() }
+    fn my_public_key(&self) -> Result<&Public, MmError<PrivKeyNotAllowed>> { utxo_common::my_public_key(self.as_ref()) }
```

Comment:
It's a bit confusing to get a `PrivKeyNotAllowed` error from `my_public_key`. As it seems related to the derivation mode, the `DerivationModeNotSupported` should be optimal here.

Reply:
You are right, `PrivKeyNotAllowed` is a special case. The problem is we don't have a derivation path to get the pubkey. Will change it.

Reply:
Done